### PR TITLE
Adjust implementation of LookupPosition.IsInMethodDeclaration to consider an expression body even in presence of a block.

### LINKED ISSUE #1702: Changing expression body member to normal method crashes VS
Here is the repo:

- Open Roslyn.sln in Visual Studio. 
- Open https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/csc/Program.cs
- Put the cursor at the end of the line that has the `Main` identifier
- Type the following: `<Enter>{`
    - Typing `{` will cause a brace completion `}` 

At this point either VS will crash immediately or it will wait until you move the cursor.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixed #1702.

@VSadov, @agocke, @gafter, @jaredpar Please review.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/ExpressionBodiedMemberTests.cs [hunk 1]
```diff
@@ -382,5 +382,226 @@ class Program
...
+            comp.VerifyDiagnostics(
+    // (8,5): error CS1519: Invalid token '=>' in class, struct, or interface member declaration
+    //     => P1;
+    Diagnostic(ErrorCode.ERR_InvalidMemberDecl, "=>").WithArguments("=>").WithLocation(8, 5),
+    // (8,10): error CS1519: Invalid token ';' in class, struct, or interface member declaration
+    //     => P1;
+    Diagnostic(ErrorCode.ERR_InvalidMemberDecl, ";").WithArguments(";").WithLocation(8, 10),
+    // (8,10): error CS1519: Invalid token ';' in class, struct, or interface member declaration
+    //     => P1;
+    Diagnostic(ErrorCode.ERR_InvalidMemberDecl, ";").WithArguments(";").WithLocation(8, 10)
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/ExpressionBodiedMemberTests.cs]: Will open an issue for the duplicate error once this PR is merged.

**Comment by VSadov** [comment]
LGTM

**Comment by AlekseyTs** [comment]
Added more tests

File: src/Compilers/CSharp/Test/Semantic/Semantics/ExpressionBodiedMemberTests.cs [hunk 2]
```diff
@@ -382,5 +382,354 @@ class Program
                 //     private void M() => (new object());
                 Diagnostic(ErrorCode.ERR_IllegalStatement, "(new object())").WithLocation(4, 25));
         }
+
+        [Fact, WorkItem(1702, "https://github.com/dotnet/roslyn/issues/1702")]
```
Comment (gafter) [inline src/Compilers/CSharp/Test/Semantic/Semantics/ExpressionBodiedMemberTests.cs]: My sense is that the set of tests is overkill (i.e. not worth the additional execution time every time tests are run, relative to the probability of catching problems). I recommend combining them into a single test case (i.e. a single compilation).

Comment (gafter) [inline src/Compilers/CSharp/Test/Semantic/Semantics/ExpressionBodiedMemberTests.cs]: I recommend simplifying things by changing the syntax tree API by adding `ExpressionBody` to `BaseMethodDeclarationSyntax`. It looks like an oversight that it isn't that way today.

File: src/Compilers/CSharp/Test/Semantic/Semantics/ExpressionBodiedMemberTests.cs (hunk 1 above)
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/ExpressionBodiedMemberTests.cs]: Opened #2052.

**Pull request merged**